src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private final String fruit;
+    private final int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setOperation(Operation operation) {
+        this.operation = operation;
+    }
+}

```

Comment:
Consider hiding inner class elements with the help of access modifiers to follow the encapsulation principle. Currently, `fruit` and `quantity` fields are declared as `final`, but there are no access modifiers specified, which means they have package-private access by default. You should define them as `private` to ensure encapsulation.